cheatsheets/Mobile_Application_Security_Cheat_Sheet.md

```diff
@@ -246,11 +258,119 @@ examples of data that should not be logged.
+potentially bypass authentication if not secured, allowing unauthorized
+users access to secure sections of the app.
+- An example of this on Microsoft Authenticator for iOS (which was
+remediated in July 2024) allowed users to bypass App Lock by simply
+navigating to `msauth://microsoft.aad.brokerplugin/?`, which would
+open Authenticator and dismiss the Face ID/Touch ID/passcode prompt.
+- **How**: Implement authentication checks on any view controllers
+or endpoints accessed via deep links. Configure and validate Universal
+Links using apple-app-site-association files for secure deep linking.
+Sanitize and validate all parameters received through deep links to
+prevent injection attacks. Ensure unauthorized users are redirected
+to the login screen, preventing direct access to sensitive parts of
+the app without proper authentication. (See Apple Developer's
+[Supporting universal links in your app](
+https://developer.apple.com/documentation/xcode/supporting-universal-links-in-your-app)
+documentation for more information.)
+- Deep links offer direct access to specific app screens, which could potentially bypass authentication if not secured, allowing unauthorized users access to secure sections of the app.
+- An example of this on Microsoft Authenticator for iOS (which was remediated in July 2024) allowed users to bypass App Lock by simply navigating to `msauth://microsoft.aad.brokerplugin/?`, which would open Authenticator and dismiss the Face ID/Touch ID/passcode prompt.
+- **How**: Implement authentication checks on any view controllers or endpoints accessed via deep links. Configure and validate Universal Links using apple-app-site-association files for secure deep linking. Sanitize and validate all parameters received through deep links to prevent injection attacks. Ensure unauthorized users are redirected to the login screen, preventing direct access to sensitive parts of the app without proper authentication. (See Apple Developer's [Supporting universal links in your app](https://developer.apple.com/documentation/xcode/supporting-universal-links-in-your-app) documentation for more information.)
+
+#### WidgetKit Security
+
+- Widgets on the lock screen may display sensitive data, potentially
+exposing it without the device being unlocked.
+- **How**: For iOS/iPadOS versions 17 and higher, use `WidgetInfo.isLocked`
+to detect lock screen state. For earlier iOS versions, implement custom
+logic based on available widget states since `widgetFamily` alone doesn't
+directly provide lock screen information. Apply conditional logic to mask
+or restrict sensitive widget content when appropriate security conditions
+aren't met. (See Apple's [WidgetKit security](
+https://support.apple.com/guide/security/widgetkit-security-secbb0a1f9b4/web)
+for more information.)
+
+#### Additional Security Considerations
+
+- Configure appropriate background refresh policies to prevent sensitive data
+updates while the device is locked.
```

Comment:
Is this really necessary? What's the risk to having updates occur while the device is locked, if the data can't be viewed until it's unlocked? Assuming we're using TLS.

Reply:
Also, some apps might legitimately require this functionality.

Reply:
I say we drop this

Reply:
Hi all,

Thank you very much for your feedback and input. On reflection, I agree with @szh that TLS does largely mitigate the risks associated with transmitting sensitive info in background updates and @ottosulin is absolutely correct that there are many legitimate reasons to enable this. The main security risks I had in mind were cases where OAuth access/refresh tokens (or other sensitive info) might be included in URLs that are transmitted over unencrypted HTTP (which I have seen happen), but I agree we can omit this for now.